### ISSUE TITLE
The GitHub Label Cleanup Proposal

### ISSUE THREAD

**Issue by Hundrec** [FunkinCrew/Funkin#3920]
# Overview
The labeling system in this repository has become more and more useful since 0.3.0 was released. 
For example, the `status` and `type` labels are now automatically applied upon issue creation. 
However, a few inconsistencies have arisen in the past few months as new labels and GitHub actions have been implemented.

In this issue, I will detail actions that can be taken to organize and standardize labels to improve your workflow (especially with over 600 open issues!)

> [!IMPORTANT]  
> Some of these changes will require edits to files such as [label-actions.yml](https://github.com/FunkinCrew/Funkin/blob/main/.github/label-actions.yml). I have created an associated PR: https://github.com/FunkinCrew/Funkin/pull/3918.

# Delete or Replace
Some labels duplicate the function of existing labels, or should be replaced with a better label. 

| Label  | Action |
| ------------- | ------------- |
| `Bug` | Delete |
| `Fixed Interally` | Delete |
| `duplicate` | Delete |
| `potential duplicate` | Delete |
| `Needs R&D` | Replace with `status: needs r&d` |

# Categorize
The [full list of labels](https://github.com/FunkinCrew/Funkin/labels) is currently alphabetically disorganized. Sorting them into categories can clean this list up.

| Category | Label | New Name | PR |
| ------------- | ------------- | ------------- | ------------- |
| Size | `small` | `size: small` | https://github.com/FunkinCrew/Funkin/pull/3918 |
| Size | `medium` | `size: medium` | https://github.com/FunkinCrew/Funkin/pull/3918 |
| Size | `large` | `size: large` | https://github.com/FunkinCrew/Funkin/pull/3918 |
| Topic | `help wanted` | `topic: help wanted` |
| Topic | `mods` | `topic: mods` |
| Topic | `platform specific` | `topic: platform specific` |
| Topic | `polish` | `topic: polish` |
| Topic | `scoring` | `topic: scoring` |
| Topic | `chart editor` | `topic: chart editor` |
| PR | `haxe` | `pr: haxe` |
| PR | `documentation` | `pr: documentation` |

# Color
Currently, [Label Actions](https://github.com/FunkinCrew/Funkin/blob/main/.github/label-actions.yml) closes issues as not planned or complete for three `status` labels. 

Just like the label `status: rejected`, highlighting these auto-close labels with prominent colors will serve two functions:
- Allow devs to easily spot and apply labels that close the issue in the desired manner.
- Allow community members to easily spot labels that indicate resolution.

| Label | Current Color | New Color | Purpose |
| ------------- | ------------- | ------------- | ------------- |
| `status: needs r&d` | Orange | `#1B3A4D` | Group with `status` |
| `status: resolved` | `#1B3A4D` | `#238636` | Indicate auto-close |
| `status: stale` | `#1B3A4D` | `#71383A` | Indicate auto-close |
| `status: duplicate` | `#1B3A4D` | `#71383A` | Indicate invalid issue |
| `low priority` | `#ffffff` | `#c8c8c8` | White should be reserved for automatically applied PR labels | 

Additional color changes for `topic:` labels are listed below.

| Label | Current Color | New Color |
| ------------- | ------------- | ------------- |
| `topic: help wanted` | `#0e8a16` | `#FCF400` |
| `topic: mods` | `#0e8a16` | `#c24b99` |
| `topic: platform specific` | `#0e8a16` | `#c24b99` |
| `topic: polish` | `#0e8a16` | `#c24b99` |
| `topic: scoring` | `#0e8a16` | `#c24b99` |
| `topic: chart editor` | `#FCF400` | `#c24b99` |

> [!TIP]
> Refer to the table below to see which labels close issues when applied.

| Label | Close Reason |
| ------------- | ------------- |
| `status: resolved` | Completed |
| `status: rejected` | Not planned |
| `status: stale` | Not planned |
| `status: duplicate` | (no longer auto-closes) |

# Create
Our current labels don't quite cover the full range of issues and PR types. We can add some labels to help!

| Label | Color | Description |
| ------------- | ------------- | ------------- |
| `pr: github` | `#ffffff` | PR modifies GitHub files. |
| `type: github` | `#CA995E` | Relates to the management of issues and pull requests on GitHub. |

# Descriptions
Some labels do not currently have descriptions. Assuming all changes before this section have been implemented, the labels below should have their descriptions revised.

| Label | Proposed Description |
| ------------- | ------------- |
| `status: resolved internally` | Completed development and will be released some time in the future. |
| `pr: haxe` | PR modifies game code. |
| `pr: documentation` | PR modifies documentation or README files. |
| `topic: mods` | Related to the use of mods. |
| `topic: chart editor` | Related to the operation of the Chart Editor. |
| `topic: polish` | Involves minor polish to the UI or gameplay. |
| `topic: scoring` | Involves the game's ranking and scoring systems. |
| `topic: platform specific` | Specific to only one or several of the game's supported targets. |
| `type: documentation` | Involves improvements or additions to documentation. |
| `type: enhancement` | Involves an enhancement or new feature. |
| `size: small` | A small pull request with 10 or fewer changes. |
| `size: medium` | A medium pull request with 100 or fewer changes. |
| `size: large` | A large pull request with more than 100 changes. |

# Closing
> [!IMPORTANT]  
> These changes should be replicated in the [funkin.assets](https://github.com/FunkinCrew/funkin.assets/pulls) repo as well.

I believe that organization is essential for an improved workflow on GitHub.
Thank you for your time and consideration!

**Comment by AbnormalPoof** [comment]
This seems great for the most part, but I do feel like some labels (`status: needs r&d`, `topic: help wanted`, `low priority`) could be part of a misc. category of some sorts (something like `tag: `?).

**Comment by Hundrec** [comment]
Good idea! I also felt like rounding up those labels would complete the alphabetization, but wasn't sure what to call the new category.

**Comment by EliteMasterEric** [comment]
Hey @Hundrec and @AbnormalPoof do you have Discord?

**Comment by Hundrec** [comment]
Yeah! Add me @hundrec and I'll reply when I get home :)

**Comment by AbnormalPoof** [comment]
I'm only just now seeing this at 6 AM...

But yeah! My Discord username is `kagaminerinlen`.

**Comment by EliteMasterEric** [comment]
All these changes are approved and done!
